pkg/pillar/hypervisor/qmp.go

```diff
@@ -15,71 +18,80 @@ import (
 // this package implements subset of
 //     https://qemu.weilnetz.de/doc/qemu-qmp-ref.html
 
-const sockTimeout = 10 * time.Second
+const (
+	sockTimeout   = 10 * time.Second
+	qmpRetries    = 5
+	qmpRetrySleep = 3 * time.Second
+)
 
-func execRawCmd(socket, cmd string) ([]byte, error) {
-	var retry = 3
+// execRawCmd will retry (if doReply is set) for qmpRetries times sleeping
+// qmpRetrySleep between attempts
+func execRawCmd(socket, cmd string, doRetry bool) ([]byte, error) {
+	var retry int
 	logrus.Debugf("executing QMP command: %s", cmd)
 	var err error
 	var monitor *qmp.SocketMonitor
 
+	if doRetry {
+		retry = qmpRetries
+	}
 	for retry >= 0 {
-		if monitor, err = qmp.NewSocketMonitor("unix", socket, sockTimeout); err == nil {
-			break
+		if monitor, err = qmp.NewSocketMonitor("unix", socket, sockTimeout); err != nil {
+			retry = retry - 1
+			time.Sleep(qmpRetrySleep)
+			continue
 		}
-		retry = retry - 1
-		time.Sleep(time.Second)
+		if err = monitor.Connect(); err != nil {
```

Comment:
As far as I can see, we share the retry attempts between the `.NewSocketMonitor()` and `.Connect()` calls.
So, if `.NewSocketMonitor()` succeeds only on the 5th attempt, does the `.Connect()` have only one chance?  Is it ok?

Reply:
I have the same question, maybe retry can be reset after the qmp.NewSockerMonitor() call....

Reply:
Good point. Will address.

Reply:
Fixed.